jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushToRepositoryAction.kt

```diff
@@ -0,0 +1,360 @@
+        val editExtension = ExtendableTextComponent.Extension.create(
+            AllIcons.General.Inline_edit, AllIcons.General.Inline_edit_hovered,
+            message("ecr.dockerfile.configuration.edit")
+        ) {
+            runConfiguration?.let {
+                RunManager.getInstance(project).findSettings(it)?.let { settings ->
+                    RunDialog.editConfiguration(
+                        project, settings,
+                        ExecutionBundle.message("run.dashboard.edit.configuration.dialog.title")
+                    )
+                }
+            }
+        }
+
+        val browseExtension = ExtendableTextComponent.Extension.create(
+            AllIcons.General.OpenDisk, AllIcons.General.OpenDiskHover,
+            message("ecr.dockerfile.configuration.add")
+        ) {
+            val listener = object : TextBrowseFolderListener(
+                FileChooserDescriptorFactory.createSingleFileDescriptor(DockerFileType.DOCKER_FILE_TYPE),
+                project
+            ) {
+                init {
+                    myTextComponent = textComponent
+                }
+
+                override fun getInitialFile() = this@PushToEcrDialog.project.guessProjectDir()
+
+                override fun onFileChosen(chosenFile: VirtualFile) {
+                    val settings = EcrUtils.dockerRunConfigurationFromPath(this@PushToEcrDialog.project, chosenFile.presentableName, chosenFile.path)
+                    // open dialog for user
+                    RunDialog.editConfiguration(
+                        project, settings,
+                        ExecutionBundle.message("run.dashboard.edit.configuration.dialog.title")
+                    )
+                    rebuildRunConfigurationComboBoxModel(runConfigModel)
+                }
+            }
+
+            runInEdt {
+                listener.run()
+            }
+        }
+
+        // extensions from right to left
+        textComponent.setExtensions(browseExtension, editExtension)
+    }
+
+    private fun rebuildRunConfigurationComboBoxModel(model: CollectionComboBoxModel<DockerRunConfiguration>) {
+        val configs = RunManager.getInstance(project).getConfigurationsList(DockerCloudType.getRunConfigurationType())
+            .filterIsInstance<DockerRunConfiguration>()
+            .filter {
+                // there are multiple types of Docker run configurations. only accept Dockerfile for now
+                // "image" and "compose" both seem like they only make sense as run-only configurations
+                it.deploymentSource.type == DockerFileDeploymentSourceType.getInstance()
+            }
+
+        model.replaceAll(configs)
+        model.selectedItem = configs.firstOrNull()
+    }
+
+    private fun selectedRepo() = remoteRepos.selected() ?: throw IllegalStateException("repository uri was null")
+
+    suspend fun getPushRequest(): EcrPushRequest {
+        val tag = remoteTag.nullize() ?: defaultTag
```

Comment:
Should we also take into account blank? I guess a user did intend it if it is blank though

Reply:
added validation